File: 1894-merge-strings-alternately/merge-strings-alternately.c


char * mergeAlternately(char * word1, char * word2){
    int len1=strlen(word1);
    int len2=strlen(word2);
    int mlen=len1+len2;
    char *merge;
    merge=(char *)malloc((mlen+1)*sizeof(char));
    int i=0,j=0,k=0;
    while(i<len1 && j<len2)
    {
        merge[k++]=word1[i++];
        merge[k++]=word2[j++];
    }
    while(i<len1)
    {
        merge[k++]=word1[i++];
    }
    while(j<len2)
    {
        merge[k++]=word2[j++];
    }
    merge[k]='\0';
    return merge;
}

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int len1=word1.length();
        int len2=word2.length();
        int mlen=len1+len2;
        StringBuilder merge = new StringBuilder(mlen);       
        int i = 0, j = 0;
        while (i < len1 && j < len2) {
            merge.append(word1.charAt(i++));
            merge.append(word2.charAt(j++));
        }
        
        while (i < len1) {
            merge.append(word1.charAt(i++));
        }
        
        while (j < len2) {
            merge.append(word2.charAt(j++));
        }
        
        return merge.toString();
    }
}